### ISSUE TITLE
Restored endpoint is exposed before validation

### ISSUE THREAD

**Issue by tgraf** [cilium/cilium#11656]
The following has been observed:

Endpoint is being restored from disk:
```
2020-05-21T08:37:47.357178492Z level=debug msg="Found endpoint C header file" endpointID=721 file-path=/var/run/cilium/state/721/lxc_config.h subsys=endpoint
2020-05-21T08:37:47.357378519Z level=debug msg="Endpoint restoring" code=OK containerID=29a274a608 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=721 endpointState=restoring identity=4441 ipv4=10.0.0.89 ipv6="fd00::60" k8sP
odName=default/details-v1-55cf6d74d-k98wn policyRevision=0 subsys=endpoint type=0
2020-05-21T08:37:47.35886442Z level=debug msg="Restoring endpoint" endpointID=721 k8sPodName=default/details-v1-55cf6d74d-k98wn subsys=daemon
2020-05-21T08:37:47.358879574Z level=debug msg="Restoring endpoint from previous cilium instance" code=OK containerID=29a274a608 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=721 endpointState=restoring identity=4441 ipv4=
10.0.0.89 ipv6="fd00::60" k8sPodName=default/details-v1-55cf6d74d-k98wn policyRevision=0 subsys=endpoint type=0
2020-0
```

The endpoint is being regenerated after restoring:
```
2020-05-21T08:37:48.798618923Z level=info msg="New endpoint" containerID=29a274a608 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=721 identity=4441 ipv4=10.0.0.89 ipv6="fd00::60" k8sPodName=default/details-v1-55cf6d74d-k98
wn subsys=endpoint
2020-05-21T08:37:48.805732812Z level=debug msg="Regenerating endpoint" containerID=29a274a608 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=721 identity=4441 ipv4=10.0.0.89 ipv6="fd00::60" k8sPodName=default/details-v1-55c
f6d74d-k98wn reason="syncing state to host" startTime="2020-05-21 08:37:48.80558107 +0000 UTC m=+5.296922232" subsys=endpoint
2020-05-21T08:37:48.805754636Z level=debug msg="Regenerating endpoint: syncing state to host" code=OK containerID=29a274a608 da
```

The endpoint is being identified as invalid.
```
2020-05-21T08:37:48.80994416Z level=warning msg="kubernetes pod not found, deleting endpoint from endpoint manager" endpointID=721 k8sNamespace=default k8sPodName=details-v1-55cf6d74d-k98wn subsys=daemon
```

The last step is unexpected as `validateEndpoint()` should be called before an endpoint is being regenerated due to being restored.

**Comment by mrostecki** [comment]
From what I see, `validateEndpoint()` doesn't return `false` when the pod doesn't exists - it just runs a goroutine which removes the endpoint, but then if goes further and returns `true` if it goes successfully through the rest of function.

I will submit a fix soon.

**Comment by mrostecki** [comment]
I think it's fixed by #11714?
